### PR TITLE
Add COST_ENTROPY_WITH_ACTIVATION(WAIT FOR #190)

### PR DESCRIPTION
**Pull request by zhoonit**
Add `COST_ENTROPY_WITH_*` to decouple `activation_type` from `layers`

Now `COST_ENTROPY_WITH_SIGMOID` and `COST_ENTROPY_WITH_ACTIVATION` is used

`COST_ENTROPY` now autoselects by activation type of the last layer.
if it is not applicable, it fallbacks to `COST_ENTROPY_WITH_SOFTMAX`
internally in loss calculation for `COST_ENTROPY`.

**Changes proposed in this PR:**
- Add `COST_ENTROPY_WITH_*` to cost type
- Modify test to test `COST_ENTROPY_WITH_*`
- Update ini comments
- Run clang format
- Update loss layer accordingly

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

### CODE REVIEW

**Comment by kparichay** [comment]
Do we really need to expose three types of cross_entropy cost? How about if the last activation is softmax and cost is cross-entropy, just amalgamate them internally.
I am preparing another PR (will push today) for cross-entropy with sigmoid because the amalgamated calculation is numerically more stable, you can refer to that.

**Comment by zhoonit** [comment]
My intention for this PR is mainly to **scrape `activation_type`**  out of abstract `layer` class. When activation is decoupled to a layer, other layer shouldn't be depend on the `activation_type` by design. So I was to divide `COST_ENTROPY`  into `COST_ENTROPY_SIGMOID` and `COST_ENTROPY_SOFTMAX` for better API.

> Do we really need to expose three types of cross_entropy cost? How about if the last activation is softmax and cost is cross-entropy, just amalgamate them internally (internally we can have as many types as we want but IMHO exposing more types outside just complicates the API).

 You are right. I agree that it is good to provide small set of functionality as API nor this PR is intended to extend the API. I've got some thoughts about this though, but I think I should better have a discussion first before proceed. Maybe next week :).

> I am preparing another PR (#193) for cross-entropy with sigmoid because the amalgamated calculation is numerically more stable with exposing outside.

Could you consider revising #193 to not use `activation_type` inside `loss_layer`? If it is okay with you, I think I can close this PR and merge your version.

**Comment by kparichay** [comment]
Yes, I would love to have activation_type move out of regular layers. I will move the sigmoid activation move out of the loss layer.
I will move softmax out too replaced with the amalgamated more stable version in this PR with an update today.

I am sorry for causing trouble with this PR of yours.

**Comment by jijoongmoon** [comment]
it will be done in this PR #193

**Comment by zhoonit** [comment]
> Yes, I would love to have activation_type move out of regular layers. I will move the sigmoid activation move out of the loss layer.
> I will move softmax out too replaced with the amalgamated more stable version in this PR with an update today.
> 
> I am sorry for causing trouble with this PR of yours.

That's okay! Thank you for addressing this issue out :)
